README.md

```diff
@@ -99,6 +99,14 @@ struct
 |> set_rank(rank: :global_rank, position: :global_position)
 ```
 
+You can also pass a `:base_queryable` to scope more complex cases:
+
+```elixir
+struct
+|> cast(params, [:position])
+|> set_rank(base_queryable: Ecto.Query.where(queryable, [item], not is_nil(item.deleted)))
```

Comment:
I think the example can be modified to work in the classic usage (in the schema)

```suggestion
import Ecto.Query
not_deleted_query = from(items in __MODULE__, where: not is_nil(items.deleted))

struct
|> cast(params, [:position])
|> set_rank(base_queryable: not_deleted_query)
```

Reply:
I committed but actually this example is wrong.
The `base_queryable` is not a query directly, is a reference to a function that receives two params.

I'll change the example